R/zchunk_LB109.ag_an_ALL_R_C_Y.R

```diff
@@ -33,61 +33,209 @@ module_aglu_LB109.ag_an_ALL_R_C_Y_DISABLED <- function(command, ...) {
+    ag_Flow_cols <- c( "Prod_Mt", "NetExp_Mt", "Supply_Mt", "Food_Mt", "Feed_Mt", "Biofuels_Mt", "OtherUses_Mt" )
+    # List of any commodities (e.g. pasture, residue, scavenging) in feed but not in production tables
+    L103.ag_Prod_Mt_R_C_Y %>%
+      select(GCAM_commodity) %>%
+      unique() %>%
+      .[['GCAM_commodity']] -> Primary_commodities
+    L108.ag_Feed_Mt_R_C_Y %>%
+      filter(!(GCAM_commodity %in% Primary_commodities)) %>%
+      select(GCAM_commodity) %>%
+      unique() %>%
+      .[['GCAM_commodity']] -> Feed_commodities
+
+    # Combine all flow tables
+    L106.ag_NetExp_Mt_R_C_Y %>%
+      bind_rows(L108.ag_NetExp_Mt_R_FodderHerb_Y) %>%
+      # Name the flows in each table, and combine all tables
+      mutate(flow = "NetExp_Mt") %>%
+      bind_rows(mutate(L103.ag_Prod_Mt_R_C_Y, flow = "Prod_Mt")) %>%
+      bind_rows(mutate(L101.ag_Food_Mt_R_C_Y, flow = "Food_Mt")) %>%
+      bind_rows(mutate(L108.ag_Feed_Mt_R_C_Y, flow = "Feed_Mt")) %>%
+      bind_rows(mutate(L122.in_Mt_R_C_Yh, flow = "Biofuels_Mt")) %>%
+      # Get all combinations of each GCAM_commodity and flow, by spreading to wide format
+      spread(flow, value) %>%
+      # Set missing values in the complete combinations to zero
+      mutate_if(is.numeric, funs(replace(., is.na(.), 0))) %>%
+      # For any feed commodities (e.g. pasture, residue, scavenging), set production = feed
+      mutate(Prod_Mt = if_else(GCAM_commodity %in% Feed_commodities, Feed_Mt, Prod_Mt),
+             # Calculate the domestic supply
+             Supply_Mt = Prod_Mt - NetExp_Mt,
+             # Calculate other uses
+             OtherUses_Mt = Supply_Mt - Food_Mt - Feed_Mt - Biofuels_Mt) -> L109.ag_ALL_Mt_R_C_Y
+
+    # Adjust global and regional crop mass balances to remove net negative other uses
+    # Assign negative other net uses to imports, and adjust global trade to maintain balances
+    # Changes in global net exports are apportioned among regions with positive other uses, according to regional shares
+
+    # First calculate the changes in global net exports
+    if(any(L109.ag_ALL_Mt_R_C_Y$OtherUses_Mt < 0)) {
+      L109.ag_ALL_Mt_R_C_Y %>%
+        # Subset negative and positive "other uses" separately
+        mutate(NegOtherUses_Mt = if_else(OtherUses_Mt < 0, OtherUses_Mt, 0),
+               # Positive will be the new adjusted "other uses", and replace negative with zero
+               OtherUses_Mt_adj = if_else(OtherUses_Mt >= 0, OtherUses_Mt, 0),
+               # Assign negative "other uses" to imports, and calculate the adjusted regional net exports
+               NetExp_Mt_adj = NetExp_Mt + NegOtherUses_Mt) %>%
+        group_by(GCAM_commodity, year) %>%
+        # Calculate the changes in global net exports = sum of negative other uses, global other uses =  sum of positive other uses
+        summarise_at(vars(GlobalNetExpAdj = NegOtherUses_Mt, GlobalOtherUses_Mt = OtherUses_Mt_adj), sum) -> L109.ag_ALL_Mt_glbl_C_Y
```

Comment:
I kind of agree with Ben...this is easier to read as `summarize(GlobalNetExpAdj = sum(NegOtherUses_Mt, etc. )`